### ISSUE TITLE
Add possibility to pass float values or range to --parameter-scan

### ISSUE THREAD

**Issue by guilhermeleobas** [sharkdp/hyperfine#184]
When using hyperfine, I am missing the possibility of specifying a range or floating point values on `--parameter-scan`:

```
hyperfine --parameter-scan i 1 10 2 "sleep {i}"
```

or 

```
hyperfine --parameter-scan i 0.1 1.0 "sleep {i}"
```

**Comment by sharkdp** [comment]
Thank you for the feedback. What CLI would you propose for this? We can't let `--parameter-scan` take an optional fourth argument.

**Comment by guilhermeleobas** [comment]
I was thinking in use `--parameter-scan` with an optional fourth argument but since that is not possible, maybe use `--parameter-range-scan` with the same semantics of the former but with a parameter indicating the range increment.

**Comment by sharkdp** [comment]
Another option would be to keep `--parameter-scan` as it is, and add a new `--parameter-step-size <step>` (or `--step-size <step>`) option.

**Comment by sharkdp** [comment]
> edit2: Or instead of adding a floating point option, add a way to format the variable generate by `--parameter-scan` using the printf notation.

You can already do this - sort of.

With `bc`:
```bash
hyperfine -P var 1 5 'sleep $(echo "{var}/10" | bc -l)'
```

With `zsh`:
``` bash
hyperfine -S zsh -P var 1 5 'sleep $(({var}.0/10))'
```

**Comment by piyushrungta25** [comment]
@sharkdp Opened #203 to add `--parameter-step-size` argument.

**Comment by sharkdp** [comment]
@piyushrungta25 has added support for `--parameter-step-size` in #203. I think it would be great if we could also add support for floating point values, such that we can run things like:

``` bash
hyperfine --parameter-scan var 0.0 1.6 --parameter-step-size 0.1 …
```

**Comment by piyushrungta25** [comment]
> I think it would be great if we could also add support for floating point values

I see two issues with this.

1. Choosing default value of the step. Something like `0.1` as the default doesn't seem very intuitive. One thing we can do is make the step size argument required for ranges with float bounds.

2. The range iterators in std library(or even [`num`](https://docs.rs/num-iter/0.1.37/num_iter/index.html) crate) does not take float parameters. So if we are okay with rolling our own iterator, I can come up with a solution.

**Comment by sharkdp** [comment]
>     1. Choosing default value of the step. Something like `0.1` as the default doesn't seem very intuitive. One thing we can do is make the step size argument required for ranges with float bounds.

Good point. I would definitely *not* choose 0.1 as the default, yes. Requiring the step size to be set for float arguments sounds like a good idea to me!

>     2\. The range iterators in std library(or even [`num`](https://docs.rs/num-iter/0.1.37/num_iter/index.html) crate) does not take float parameters. So if we are okay with rolling our own iterator, 

I'd be okay with this, as long as it comes with proper unit tests. I could imagine that the reason for the absence in the standard library is the following unfortunate property of floating point numbers:

Imagine that someone chooses `--parameter-scan var 0.0 1.0` and a step size of `0.1`. I think that most users would expect the 1.0 boundary to be inclusive. However, due to rounding errors, a naive implementation could miss the `1.0` boundary due to `0.0 + 10 * 0.1` being too large.

I guess this could probably be circumvented by computing *N = (end - start) / stepsize* and testing if it is close to an integer number. In this special case, we could round `N` to that nearby integer and make sure that we perform exactly N incrementations (or N+1 benchmarks).

> I can come up with an implementation.

:+1:

**Comment by piyushrungta25** [comment]
> unfortunate property of floating point numbers

Perhaps a [`decimal`](https://github.com/paupino/rust-decimal) data type would be more appropriate here? This would prevent substitutions like `echo 0.30000000000000004` from happening.

**Comment by sharkdp** [comment]
This has been fully implemented by @piyushrungta25 :fireworks:

**Comment by sharkdp** [comment]
Released in [hyperfine v1.7.0](https://github.com/sharkdp/hyperfine/releases/tag/v1.7.0)
